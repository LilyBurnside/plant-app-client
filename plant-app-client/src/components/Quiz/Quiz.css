.quiz {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #DBFFB7;
  border-radius: 15px;
  margin: 100px 50px 0px;
}

.quiz > h1 {
  display: flex;
  justify-content: center;
  background: none;
  margin-left: 20px;
  margin-right: 20px;
  text-align: center;
}

.answer-list {
  display: flex;
  flex-direction: column;
  max-height: 100%;
  text-align: center;
  background: none;
}

.answer-list > label { 
  display: block;
  position: relative;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  background: none;
}

.quiz-submit, .quiz-next {
  color: black;
  text-decoration: none;
  background: white;
  padding: 10px 20px;
  border: 4px solid black;
  margin: 50px;
  border-radius: 15px;
}

.answer-label{
  position: relative;
}

.answer-label:hover {
  color: #D58BFA;
  /* padding-left: 5px;
  transition: all 0.25s linear; */
}

.answer-label:checked {
  color: #D58BFA;
}

input[type=radio] {
  position: absolute;
  visibility: hidden;
}

i {
  display: inline-block;
  width: 15px;
  height: 15px;
  border-radius: 50%;
  margin-right: 5px;
  vertical-align: middle;
  border: 5px solid white;
  transition: .25s;
}

.answer-option {
  margin-right: 20px;
}

.answer-option:checked + i {
  background: #D58BFA;
}
