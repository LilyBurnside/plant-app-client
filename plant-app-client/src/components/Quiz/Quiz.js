import React from 'react'
import PlantsApiService from '../../services/plants-api-service'
//import { Link, useHistory } from 'react-router-dom'
import QuizContext from '../../contexts/QuizContext'
import './Quiz.css'

export default class Quiz extends React.Component {

  static contextType = QuizContext 

  componentDidMount() {
    this.context.clearError()
    PlantsApiService.getQuestions()
      .then(this.context.setQuestions)
      .catch(this.context.setError)
  }

  componentWillUnmount() {
    this.context.resetQuestionId()
  }

  handleNextPress = e => {
    e.preventDefault()
    this.context.addAnswer(e.target.answer.value)
    this.context.setQuestionId()
    if (this.context.questionId === 5) {
      this.props.history.push('/results')
    }
  }

  renderAnswers = question => {
    const answers = []
    for(let i = 1; i <= 4; i++) {
      if (question['answer_' + i]) {
      answers.push(
        <label key={i} htmlFor="answer">
          <input 
            type="radio" 
            className="answer-option" 
            name="answer" 
            value={i} 
          />
            {question['answer_' + i]}
        </label>)
      }
    }
    return answers
  }

  render(){
    const { error, questions } = this.context
    const question = questions.find(q => q.id === this.context.questionId)

    let button
    if (error) {
      return (error.error === `Question doesn't exist`)
        ? <p className="red">Question not found</p>
        : <p className="red">There was an error</p>
    } else if (!question) {
      return <div className="loading">loading...</div>
    } else if (question.id === 5) {
      button = <button className="quiz-submit" type="submit" >Submit Quiz</button>
    } else {
      button = <button className="quiz-next" type="submit" >Next</button>
    } 
    return(
      <div className="quiz">
       <h1>{question.question}</h1>
       <form className="answer-list" onSubmit={this.handleNextPress}>
        {this.renderAnswers(question)}
        {button}
       </form>
      </div>
    )
  }
}