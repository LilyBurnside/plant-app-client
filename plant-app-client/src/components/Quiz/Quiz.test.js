import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Quiz from './Quiz';

describe('Quiz Component', () => {

  const Context = React.createContext({questions: [{ question: 'Sample Q', answer_1: 'answer1', answer_2: 'answer2', answer_3: 'answer3', answer_4: 'answer4' }]})

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Quiz/>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  
  it('renders a question', () => {
    const wrapper = shallow(
      <Context.Provider>
        <Quiz />
      </Context.Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot()
  })

});