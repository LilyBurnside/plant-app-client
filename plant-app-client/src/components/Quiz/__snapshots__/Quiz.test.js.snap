// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Component renders a question 1`] = `<Quiz />`;
